controlplane/internal/controller/agentcontrolplane_controller_test.go

```diff
@@ -127,45 +110,28 @@ var _ = Describe("AgentControlPlane Controller", func() {
 
 		It("should add a finalizer to the agent control plane if it's not being deleted", func() {
 			By("setting the owner ref on the agent control plane")
-			agentControlPlane := &controlplanev1alpha1.AgentControlPlane{}
-			err := k8sClient.Get(ctx, typeNamespacedName, agentControlPlane)
-			Expect(err).NotTo(HaveOccurred())
 
+			agentControlPlane := getAgentControlPlane()
 			agentControlPlane.SetOwnerReferences([]metav1.OwnerReference{*metav1.NewControllerRef(cluster, clusterv1.GroupVersion.WithKind(clusterv1.ClusterKind))})
-			Expect(k8sClient.Update(ctx, agentControlPlane)).To(Succeed())
+			Expect(k8sClient.Create(ctx, agentControlPlane)).To(Succeed())
+
+			_, err := controllerReconciler.Reconcile(ctx, reconcile.Request{NamespacedName: typeNamespacedName})
+			Expect(err).NotTo(HaveOccurred())
 
 			By("checking if the agent control plane has the finalizer")
 			err = k8sClient.Get(ctx, typeNamespacedName, agentControlPlane)
 			Expect(err).NotTo(HaveOccurred())
 			Expect(agentControlPlane.Finalizers).NotTo(BeEmpty())
 			Expect(agentControlPlane.Finalizers).To(ContainElement(acpFinalizer))
 		})
-
```

Comment:
Unfortunately this type of test has several issues with fake client, around deletion and how long this object is present after deletion (cannot verify finalizer is being removed).

I'd suggest let's run these kind of tests with EnvTest, in a later stage